geoapps/base/dash_application.py

```diff
@@ -300,3 +313,313 @@ def workspace(self, workspace):
+            os.mkdir(temp_dir)
+            temp_workspace = Workspace(Path(temp_dir) / temp_geoh5)
+
+            # Update ui.json with temp_workspace to pass initialize scatter plot app
+            param_dict = self.params.to_dict()
+            param_dict["geoh5"] = temp_workspace
+
+            with fetch_active_workspace(temp_workspace):
+                with fetch_active_workspace(self.workspace):
+                    param_dict["objects"] = obj.copy(
+                        parent=temp_workspace, copy_children=True
+                    )
+
+            new_params = self.param_class(**param_dict)
+
+            ui_json_path = Path(temp_dir) / temp_geoh5.replace(".geoh5", ".ui.json")
+            new_params.write_input_file(
+                name=temp_geoh5.replace(".geoh5", ".ui.json"),
+                path=temp_dir,
+                validate=False,
+            )
+            ifile = InputFile.read_ui_json(ui_json_path)
+            ui_json_data = ifile.demote(ifile.data)
+
+            # Start server
+            port = BaseDashApplication.get_port()
+            threading.Thread(
+                target=self.start_server,
+                args=(port, self.app_class, ifile, ui_json_data, new_params),
+                daemon=True,
+            ).start()
+
+            # Make Qt window
+            self.make_qt_window(self.app_name, port)
+        return ""
+
+    @staticmethod
+    def run(app_name: str, app_class: BaseDashApplication, ui_json: InputFile):
+        """
+        Launch Qt app from terminal.
+
+        :param app_name: Name of app to display as Qt window title.
+        :param app_class: Type of app to create.
+        :param ui_json: Input file to pass to app for initialization.
+        """
+        # Start server
+        port = ObjectSelection.get_port()
+        threading.Thread(
+            target=ObjectSelection.start_server,
+            args=(port, app_class, ui_json),
+            daemon=True,
+        ).start()
+
+        # Make Qt window
+        ObjectSelection.make_qt_window(app_name, port)
+
+    @property
+    def app_name(self) -> str | None:
+        """
+        Name of app that appears as Qt window title.
+        """
+        return self._app_name
+
+    @app_name.setter
+    def app_name(self, val):
+        if isinstance(val, str) | (val is None):
+            self._app_name = val
+        else:
+            raise TypeError
+
+    @property
+    def app_class(self) -> type[BaseDashApplication]:
+        """
+        The kind of app to launch.
+        """
+        return self._app_class
+
+    @app_class.setter
+    def app_class(self, val):
+        if issubclass(val, BaseDashApplication):
+            self._app_class = val
+        else:
+            raise TypeError
+
+    @property
+    def param_class(self) -> type[BaseParams]:
+        """
+        The param class associated with the launched app.
+        """
+        return self._param_class
```

Comment:
The private attribute should be defined in the init though. I am surprised that mypy didn't complain about this.